config/PropertyNames.xml

```diff
@@ -1,255 +1,61 @@
-code similar to the following for the C++ header and source file:
-
-//
-// Properties.h
-//
-namespace IceInternal
-{
-
-//
-// ... definitions for IceInternal::Property and IceInternal::PropertyArray
-// omitted...
-
-class Properties
-{
-public:
-
-    static const PropertyArray const IceProps;
-    static const PropertyArray const IceBoxProps;
-
-    static const PropertyArray validProps[];
-};
-
-}
-
-// Properties.cpp
-const IceInternal::Property IceInternal::Properties::IcePropsData[] =
-{
-    IceInternal::Property("Ice.Package.*", false, 0),
-    IceInternal::Property("Ice.QuenchRequests", true, 0),
-    IceInternal::Property("Ice.Foo", true, "Ice.Bar"),
-    IceInternal::Property("Ice.Bar", false, 0),
-    IceInternal::Property("Ice.Default.Locator", false, 0),
-    IceInternal::Property("Ice.Default.Locator.EndpointSelection", false, 0),
-    IceInternal::Property("Ice.Default.Locator.ConnectionCached", false, 0),
-    IceInternal::Property("Ice.Default.Locator.PreferSecure", false, 0),
-    IceInternal::Property("Ice.Default.Locator.LocatorCacheTimeout", false, 0),
-    IceInternal::Property("Ice.Default.Locator.Locator", false, 0),
-    IceInternal::Property("Ice.Default.Locator.Router", false, 0),
-    IceInternal::Property("Ice.Default.Locator.CollocationOptimization", false, 0),
-    IceInternal::Property("Ice.ThreadPool.Client.Size", false, 0),
-    IceInternal::Property("Ice.ThreadPool.Client.SizeMax", false, 0),
-    IceInternal::Property("Ice.ThreadPool.Client.SizeWarn", false, 0),
-    IceInternal::Property("Ice.ThreadPool.Client.StackSize", false, 0)
-};
-
-const IceInternal::PropertyArray
-    IceInternal::Properties::IceProps(IcePropsData, sizeof(IcePropsData) /
-        sizeof(IcePropsData[0]));
-
-const IceInternal::Property IceInternal::Properties::IceBoxPropsData[] =
-{
-    IceInternal::Property("IceBox.LoadOrder", false, 0)
-};
-
-const IceInternal::PropertyArray
-    IceInternal::Properties::IceBoxProps(IceBoxPropsData, sizeof(IceBoxPropsData) /
-        sizeof(IceBoxPropsData[0]));
-
-const PropertyArray IceInternal::Properties::validProps[] =
-{
-    IceProps,
-    IceBoxProps,
-    IceInternal::PropertyArray(0, 0)
-};
-
-Note that the class name ("Properties" in this example) is generated
-from the file name, and the first component of the property name is
-generated from the section label.
-
--->
 <properties>
-    <class name="proxy" prefix-only="false">
-        <suffix name="EndpointSelection" languages="all" />
-        <suffix name="ConnectionCached" languages="all" />
-        <suffix name="PreferSecure" languages="all" />
-        <suffix name="LocatorCacheTimeout" languages="all" />
-        <suffix name="InvocationTimeout" languages="all" />
-        <suffix name="Locator" languages="all" />
-        <suffix name="Router" languages="all"/>
+    <class name="Proxy" prefix-only="false">
+        <property name="EndpointSelection" languages="all" />
```

Comment:
Renamed suffix to property as we now parse classes the standard way. Also renamed the class names to be CamelCase.